background.ts

```diff
@@ -128,6 +128,13 @@ mainEvents.on('settings-update', async(newSettings) => {
     await pathManager.enforce();
   }
 
+  if (newSettings.hideNotificationIcon) {
+    Tray.getInstance().destroy();
+  } else {
+    Tray.getInstance().setup();
+    mainEvents.emit('k8s-check-state', k8smanager);
+  }
```

Comment:
API-wise, I think something like `Tray.update(newSettings.hideNotification)` would be clearer (that calling `setup()` twice does not make a new tray), but then I guess it wouldn't lend to conditionally triggering `k8s-check-state` well.

Maybe just call it `Tray.show()` / `Tray.hide()` (as in, static methods, not instance methods)? The rest of the work doesn't really need to know that there's multiple instances.

Reply:
I like `show()`/`hide()`, but converting to static methods has a lot of knock-on effects because these methods do rely on other members in the Tray class. Perhaps my motivations for some of these Tray modifications were a bit misguided and need to be rolled back?